src/Api/Vault/AuthorizationHandlers/Collections/BulkCollectionAuthorizationHandler.cs

```diff
@@ -96,17 +100,22 @@ public class BulkCollectionAuthorizationHandler : BulkAuthorizationHandler<BulkC
     private async Task CanCreateAsync(AuthorizationHandlerContext context, IAuthorizationRequirement requirement,
         CurrentContextOrganization? org)
     {
-        // If the limit collection management setting is disabled, allow any user to create collections
-        // Otherwise, Owners, Admins, and users with CreateNewCollections permission can always create collections
+        // Owners, Admins, and users with CreateNewCollections permission can always create collections
         if (org is
-        { LimitCollectionCreationDeletion: false } or
         { Type: OrganizationUserType.Owner or OrganizationUserType.Admin } or
         { Permissions.CreateNewCollections: true })
         {
             context.Succeed(requirement);
             return;
         }
 
+        // If the limit collection management setting is disabled, allow any user to create collections
+        if (await GetOrganizationAbilityAsync() is { LimitCollectionCreationDeletion: false })
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
```

Comment:
Just noticed this - you also have to check `org != null` here, otherwise you're calling `context.Succeed` only on the basis of the collection management setting, without actually knowing whether or not the user is a member of that org. It might be worth adding a comment noting this as well, it's easy to overlook.

Please also check any other usages, including in other handlers, to make sure this is covered.

Maybe you could pass the `currentContextOrganization` into `GetOrganizationAbilityAsync`, just to make this link more explicit.

Reply:
Great catch - thank you!